storage.bs

```diff
@@ -436,28 +475,44 @@ evaluating quotas.
 
 
 
-<h2 id=ui-guidelines>User Interface Guidelines</h2>
+<h2 id=management>Management</h2>
+
+<p>Whenever a <a>storage bucket</a> is cleared by the user agent, it must be cleared in its
+entirety. User agents should avoid clearing <a>storage bucket</a> while script that is able to
+access them is running, unless instructed otherwise by the user.
+
+<p>If removal of <a>storage buckets</a> leaves the encompassing <a>storage shelf</a>'s
+<a>bucket map</a> <a for=map lt="is empty">empty</a>, then <a for=map>remove</a> that
+<a>storage shelf</a> and corresponding <a>storage key</a> from the encompassing
+<a for=/>storage shed</a>.
+
+
+<h3 id=storage-pressure>Storage pressure</h3>
+
+<p>A user agent that comes under storage pressure should clear network state and
+<a>local storage buckets</a> whose <a for="local storage bucket">mode</a> is
+"<code>best-effort</code>", ideally prioritizing removal in a manner that least impacts the user.
+
+<p>If a user agent continues to be under storage pressure, then the user agent should inform the
+user and offer a way to clear the remaining <a>local storage buckets</a>, i.e., those whose
+<a for="local storage bucket">mode</a> is "<code>persistent</code>".
+
+<p><a>Session storage buckets</a> must be cleared as browsing sessions are closed.
```

Comment:
Ah, browsing session again. This time without an XXX box.

I'm starting to feel that we should add a `<dfn>` to HTML that is mostly a big XXX box, so that we can link to that single XXX box from everywhere.